test/Microsoft.AspNetCore.Mvc.Razor.Test/TagHelpers/TagHelperComponentPropertyActivatorTest.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.AspNetCore.Mvc.ViewEngines;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+using Microsoft.AspNetCore.Routing;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Test.TagHelpers
+{
+    public class TagHelperComponentPropertyActivatorTest
+    {
+        [Fact]
+        public void InitializesViewContext()
+        {
+            // Arrange
+            var tagHelperComponent = new TestTagHelperComponent();
+            var viewContext = CreateViewContext(new DefaultHttpContext());
+
+            var contextInitializer = new TagHelperComponentPropertyActivator();
+
+            // Act
+            contextInitializer.Activate(viewContext, tagHelperComponent);
+
+            // Assert
+            Assert.Same(viewContext, tagHelperComponent.ViewContext);
+        }
+
+        private class TestTagHelperComponent : ITagHelperComponent
+        {
+            public int Order => 1;
+
+            [ViewContext]
+            public ViewContext ViewContext { get; set; }
+
+            public void Init(TagHelperContext context)
+            {
+            }
+
+            public Task ProcessAsync(TagHelperContext context, TagHelperOutput output)
+            {
+                return Task.CompletedTask;
+            }
+        }
+
+        private static ViewContext CreateViewContext(HttpContext httpContext)
```

Comment:
Use `Mock.Of<ViewContext>()` instead of having this method. When testing a complex type like `ViewContext` if you don't need all of its inner workings then no need to build it all out 😄